doc/nrf/releases/release-notes-2.4.0.rst

```diff
@@ -0,0 +1,1038 @@
+
+* :ref:`nrf_desktop_config_channel_script`:
+
+  * Added:
+
+    * Support for the device information (``devinfo``) option fetching.
+      The option provides device's Vendor ID, Product ID and generation.
+    * Support for devices using MCUboot bootloader built in the direct-xip mode (``MCUBOOT+XIP``).
+      In this mode, the image is booted directly from the secondary slot without moving it to the primary slot.
+
+MCUboot
+=======
+
+The MCUboot fork in |NCS| (``sdk-mcuboot``) contains all commits from the upstream MCUboot repository up to and including ``6902abba270c0fbcbe8ee3bb56fe39bc9acc2774``, with some |NCS| specific additions.
+
+The code for integrating MCUboot into |NCS| is located in the :file:`ncs/nrf/modules/mcuboot` folder.
+
+The following list summarizes both the main changes inherited from upstream MCUboot and the main changes applied to the |NCS| specific additions:
+
+* Added:
+
+  * Support for the downgrade prevention feature using hardware security counters (:kconfig:option:`CONFIG_MCUBOOT_HARDWARE_DOWNGRADE_PREVENTION`).
+  * Generation of a new variant of the :file:`dfu_application.zip` when the :kconfig:option:`CONFIG_BOOT_BUILD_DIRECT_XIP_VARIANT` Kconfig option is enabled.
+    This archive file contains images for both slots, primary and secondary.
+  * Encoding of the image start address into the header when the :kconfig:option:`CONFIG_BOOT_BUILD_DIRECT_XIP_VARIANT` Kconfig option is enabled.
+    The encoding is done using the ``--rom-fixed`` argument of the :file:`imgtool.py` script.
+    If the currently running application also has the :kconfig:option:`CONFIG_MCUMGR_GRP_IMG_REJECT_DIRECT_XIP_MISMATCHED_SLOT` Kconfig option enabled, the MCUmgr rejects application image updates signed without the start address.
+
+Zephyr
+======
+
+.. NOTE TO MAINTAINERS: All the Zephyr commits in the below git commands must be handled specially after each upmerge and each nRF Connect SDK release.
+
+The Zephyr fork in |NCS| (``sdk-zephyr``) contains all commits from the upstream Zephyr repository up to and including ``4bbd91a9083a588002d4397577863e0c54ba7038``, with some |NCS| specific additions.
+
+For the list of upstream Zephyr commits (not including cherry-picked commits) incorporated into nRF Connect SDK since the most recent release, run the following command from the :file:`ncs/zephyr` repository (after running ``west update``):
+
+.. code-block:: none
+
+   git log --oneline 4bbd91a908 ^fcaa60a99f
+
+For the list of |NCS| specific commits, including commits cherry-picked from upstream, run:
+
+.. code-block:: none
+
+   git log --oneline manifest-rev ^4bbd91a908
+
+The current |NCS| main branch is based on revision ``4bbd91a908`` of Zephyr.
+
+Additions specific to |NCS|
+---------------------------
+
+* Updated TF-M version to 1.7.0.
+
+zcbor
+=====
+
+|no_changes_yet_note|
+
+Trusted Firmware-M
+==================
+
+|no_changes_yet_note|
+
+cJSON
```

Comment:
@mbolivar-nordic, is there anything to add here?

Reply:
No changes. The cjson repository has not changed at all since v2.3.0.